jwst/outlier_detection/outlier_detection.py

```diff
@@ -26,9 +29,97 @@
 
 
 __all__ = ["OutlierDetection", "flag_cr", "abs_deriv"]
+    
+
+def modelwise_operation(func: Callable) -> Callable:
+    """
+    Decorator to facilitate computations over a ModelContainer
+    preserving the spatial axis but splitting up the time axis, 
+    i.e., one ImageModel at at time
+
+    Notes
+    -----
+    Input function first two parameters MUST be self, datamodel
+    """
+    @functools.wraps(func)
+    def wrapper(self, container: ModelContainer, *args, **kwargs):
+        if not isinstance(container, ModelContainer):
+            raise ValueError("Input must be a ModelContainer")
+        
+        # store original value of return_open, then turn it off for now
+        ropen_orig = container._return_open
+        container._return_open = False
+
+        result_list = []
+        for i, model in enumerate(container):
+
+            # perform the requested operation
+            model = datamodel_open(model)
```

Comment:
Is the expectation that `container` will contain filenames? If so setting `_return_open=True` would allow a bit of simplification:
```suggestion
        # store original value of return_open, then turn it off for now
        ropen_orig = container._return_open
        container._return_open = True

        result_list = []
        for i, model in enumerate(container):
            # perform the requested operation
```